includes/admin/class-rop-admin.php

```diff
@@ -1003,6 +1005,76 @@ public function rop_dismiss_cron_disabled_notice() {
 
 	}
 
+	/**
+	 * Migrate the taxonomies from General Settings to Post Format for Pro users.
+	 * @since 8.5.4
+	 */
+	public function migrate_taxonomies_to_post_format() {
+
+		$global_settings = new Rop_Global_Settings();
+		// If any type of Pro is installed and active.
+		if ( $global_settings->license_type() > 0 ) {
+			// Get the current plugin options.
+			$option = get_option( 'rop_data' );
+			// Get the custom options.
+			// If this option exists, then the migration took place, and it will not happen again.
+			// Should return false the first time as it does not exist.
+			$update_took_place = get_option( 'rop_data_migrated_tax' );
+
+			$general_settings = array();
+			// Making sure the option we need, exists.
+			if ( empty( $update_took_place ) && isset( $option['general_settings'] ) ) {
+				$general_settings = $option['general_settings'];
+
+				$selected_taxonomies = array();
+				$exclude_taxonomies  = '';
+				if ( isset( $general_settings['selected_taxonomies'] ) ) {
+					// Get the selected Taxonomies from General Settings tab.
+					$selected_taxonomies = $general_settings['selected_taxonomies'];
+				}
+
+				// Making sure to check "Excluded" if the main General Tab ahs it checked.
+				if ( isset( $general_settings['exclude_taxonomies'] ) && ! empty( $general_settings['exclude_taxonomies'] ) ) {
+					$exclude_taxonomies = $general_settings['exclude_taxonomies'];
+				}
+
+				// If there are any taxonomies selected in the general tab.
+				if ( ! empty( $selected_taxonomies ) ) {
+
+					if ( isset( $option['post_format'] ) && ! empty( $option['post_format'] ) ) {
+						foreach ( $option['post_format'] as &$social_media_account_data ) {
+							// If the options exists in Post Format but it's empty or,
+							// If the option does not exist at all.
+							if (
+								! isset( $social_media_account_data['taxonomy_filter'] ) ||
+								(
+									isset( $social_media_account_data['taxonomy_filter'] ) &&
+									empty( $social_media_account_data['taxonomy_filter'] )
+								)
+							) {
+								// Add the taxonomies to all social media accounts.
+								$social_media_account_data['taxonomy_filter'] = $selected_taxonomies;
+								// inform that the update took place.
+								$update_took_place = true;
```

Comment:
Wouldn't this be the last line of this operation? I noticed after this we're still setting some array values for exclude taxonomies

Reply:
If you are referring to this line
`$social_media_account_data['exclude_taxonomies'] = $exclude_taxonomies;` 
This is for the checkbox "Exclude", if it's checked in the General settings, we transfer this to Each social media accounts also.

taxonomies and the checkbox are set as 2 separate variables and values.